### PR TITLE
WIP:imageset job metrics

### PR DESCRIPTION
**Pull request by ShivamChamoli**
imageset job metrics

### CODE REVIEW

File: pkg/controller/clusterdeployment/clusterdeployment_controller.go [hunk 1]
```diff
@@ -109,13 +109,29 @@ var (
 			Buckets: []float64{30, 60, 120, 300, 600, 1200, 1800},
 		},
 	)
+	imagesetJobsDurationHistogram = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "imageset_jobs_duration_seconds",
```
Comment (dgoodwin) [inline pkg/controller/clusterdeployment/clusterdeployment_controller.go]: All our metrics should be prefixed with hive_. Could you update this and the ones above that aren't?

**Comment by ShivamChamoli** [comment]
--> Kickstart is giving a weirdly high number (>600) although it launches almost immediately
--> Not sure why the two gauges in metrics.go are always 0
--> As for the duration of the job I need to return a boolean from resolveInstallerImage function which tells me that the job has completed, is it ok to change the function signature to send a boolean along with an error

File: pkg/controller/clusterdeployment/clusterdeployment_controller.go [hunk 2]
```diff
@@ -571,6 +587,12 @@ func (r *ReconcileClusterDeployment) resolveInstallerImage(cd *hivev1.ClusterDep
 		err = r.Create(context.TODO(), job)
 		if err != nil {
 			jobLog.WithError(err).Error("error creating job")
+		} else {
+			// kickstartDuration calculates the delay between creation of cd and start of imageset job
+			kickstartDuration := time.Since(cd.CreationTimestamp.Time)
+			cdLog.Infof("KICKSTART:%v", kickstartDuration.Seconds())
```
Comment (dgoodwin) [inline pkg/controller/clusterdeployment/clusterdeployment_controller.go]: Drop this line please.

File: pkg/controller/metrics/metrics.go [hunk 3]
```diff
@@ -210,6 +222,32 @@ func (mc *Calculator) Start(stopCh <-chan struct{}) error {
 			}
 		}
 
+		mcLog.Info("IMAGESET JOBS")
```
Comment (dgoodwin) [inline pkg/controller/metrics/metrics.go]: Drop this line please.

File: pkg/imageset/generate.go [hunk 4]
```diff
@@ -47,6 +47,11 @@ else
   echo "0" > /common/success
 fi
 `
+	// ImagesetJobLabel is the label used for counting the number of imageset jobs in Hive
+	ImagesetJobLabel = "hive.openshift.io/imageset"
+
+	// ClusterDeploymentNameLabel is the label (along with ClusterDeploymentNamespaceLabel) that is used to identify the imageset pod of a particular cluster deployment
```
Comment (dgoodwin) [inline pkg/imageset/generate.go]: Can drop the note about ClusterDeploymentNamespaceLabel as we dropped that.

File: pkg/controller/clusterdeployment/clusterdeployment_controller.go (hunk 2 above)
Comment (dgoodwin) [reply pkg/controller/clusterdeployment/clusterdeployment_controller.go]: So this is showing over 600 despite it starting immediately? hmmm. I can't see how that would happen...

**Comment by dgoodwin** [comment]
> --> Kickstart is giving a weirdly high number (>600) although it launches almost immediately
> --> Not sure why the two gauges in metrics.go are always 0

I have a suspicion this is because they get cleaned up once they're successful, so you would only see  them if you listed metrics *while* the job is running. This is probably working as intended.

> --> As for the duration of the job I need to return a boolean from resolveInstallerImage function which tells me that the job has completed, is it ok to change the function signature to send a boolean along with an error

Little confused, can you clarify why you need to return a boolean for if the job has completed or not?

**Pull request merged**